# <font color='darkcyan'>Practice Sheet on String and List</font>                

**1.** Write a python code of a program that reads the values for the three sides x, y, and z of a triangle,
and then calculates its area. . The area is calculated as follows:

**Area =** $s\sqrt{(s-x)(s-y)(s-z)} \mbox{   }$  **where,**   s=$\frac{x+y+z}{3}$

In [11]:
#todo
x=int(input('Enter the value of X: '))
y=int(input('Enter the value of Y: '))
z=int(input('Enter the value of Z: '))

s=(x+y+z)/3
area=s*(((s-x)*(s-y)*(s-z))**0.5)
print(s)
print(area)

Enter the value of X: 2
Enter the value of Y: 3
Enter the value of Z: 6
3.6666666666666665
(3.615092585593557e-16+5.90389422992903j)


**2.** Write Python code of a program that reads three numbers form user and print the greatest number
among three numbers.


In [15]:
#todo
x=int(input('Enter the value of X: '))
y=int(input('Enter the value of Y: '))
z=int(input('Enter the value of Z: '))

if(x>y and x>z):
    print('Greatest',x)
elif(y>x and y>z):
    print('Greatest',y)
else:
    print('Greatest',z)


Enter the value of X: 3
Enter the value of Y: 4
Enter the value of Z: 5
Greatest 5


**3.** Write Python code of a program that reads three sides of a triangle and check whether the triangle is
valid or not.

In [4]:
#todo
x=int(input('Enter the value of X: '))
y=int(input('Enter the value of Y: '))
z=int(input('Enter the value of Z: '))

list1=[x,y,z]
max_value=max(list1)
total_sum= x+y+z-max_value

if(total_sum > max_value):
    print('Valid')
else:
    print('Invalid')

Enter the value of X: 3
Enter the value of Y: 4
Enter the value of Z: 8
Unvalid


**4.** Three conditions are used to identify leap years:

I. The year can be evenly divided by 4, is a leap year, unless

II. The year can be evenly divided by 100, it is NOT a leap year, unless:

III. The year is also evenly divisible by 400. Then it is a leap year.
#### 
For example, the years 2000 and 2400 are leap years, while 1800, 1900, 2100, 2200, 2300 and 2500 are
NOT leap years.

Write Python code of a program that reads a number as year and determine whether it is a leap year or not.

In [2]:
#todo


**5.** Write a Python program that reads a number and find the sum of the series of

1 +11 + 111 + 1111 + ….+N terms.

Example:

**Sample Input :**

5

**Sample Output :**

1 + 11 + 111 + 1111 + 11111


The Sum is : 12345


In [1]:
#todo
number=int(input('Enter the number: '))

count='1'
for i in range(0,number):
    print(count,end=' + ')
    count= count + '1'

Enter the number: 5
1 + 11 + 111 + 1111 + 11111 + 

**6.** Write a Python program that reads a number and display the multiplication table of the given integer.

Example :

**Sample Input :** 15

**Sample Output :**

15 X 1 = 15

15 X 2 = 30

**...**

**...**

15 X 10 = 150

In [3]:
#todo
number=int(input('Enter the number: '))

for i in range(1,11):
    num= number*i
    print(number,'*',i,'=',num)

Enter the number: 15
15 * 1 = 15
15 * 2 = 30
15 * 3 = 45
15 * 4 = 60
15 * 5 = 75
15 * 6 = 90
15 * 7 = 105
15 * 8 = 120
15 * 9 = 135
15 * 10 = 150


**8.** Write a python program that takes integer inputs from the user until the user gives "Stop". Print
average and product of all numbers

In [5]:
#todo
number=input('Enter the number: ')
list1=[]

while number!= 'Stop':
    list1.append(int(number))
    number=input('Enter the numbers: ')

sum1=0
product= 1
for i in list1:
    sum1+= i
    product= product*i
    
avg= sum1/len(list1)


print(avg)
print(product)


Enter the number: 5
Enter the numbers: 6
Enter the numbers: 7
Enter the numbers: Stop
6.0
210


**9.** Write a Python program to count the total number of alphabets, digits and special characters in a
string.

**Sample Input:**

Welcome to geeksforgeeks.com

**Sample Output:**

Number of Alphabets in the string is : 25

Number of Digits in the string is : 0

Number of Special characters in the string is : 3

In [12]:
#todo
sentence=input('Enter the sentence: ')

count_alp=0
count_digit=0
count_sp=0
for i in sentence:
    x=ord(i)
    if(x >= 65 and x <= 122):
        count_alp += 1
    elif(x >= 48 and x<= 57):
        count_digit += 1
    else:
        count_sp += 1

print('Number of Alphabets in the string is : ',count_alp)
print('Number of Digits in the string is : ',count_digit)
print('Number of Special characters in the string is : ',count_sp)


Enter the sentence: Welcome to geeksforgeeks.com
Number of Alphabets in the string is :  25
Number of Digits in the string is :  0
Number of Special characters in the string is :  3


**10.** Write a Python program to find the largest and smallest word in a string.

**Sample Input :**

It is a string with the smallest and largest word.

**Sample Output :**

The largest word is “smallest”

and the smallest word is 'a'

in the string : 'It is a string with the smallest and largest word.'.

In [19]:
#todo
sentence=input('Enter the Sentence: ')
list1=sentence.split(' ')

max_st =list1[0]
min_st =list1[0]

for i in range(0,len(list1)):
    if(len(max_st) < len(list1[i])):
        max_st = list1[i]
    if(len(min_st) > len(list1[i])):
        min_st = list1[i]

print('The largest word is: ',max_st)
print('and the smallest word is: ',min_st)
print('in the string : ',sentence)

Enter the Sentence: It is a string with the smallest and largest word.
The largest word is:  smallest
and the smallest word is:  a
in the string :  It is a string with the smallest and largest word.


**11.** Write a Python program that takes two strings from the user . Then remove characters from the first

string which are present in the second string.

**<font color='red'>[ Avoid built-in function]</font>**

**Sample Input:**

India is great

in

**Sample Output:**

da s great

In [35]:
#todo
word1=input('Enter the first word: ')
word2=input('Enter the second word: ')
list1=list(word1)
list2=list(word2)

for k in list2:
    if(k in list1):
        list1.remove(k)
        
for i in list1:       
    print(i,end='')

Enter the first word: India is great
Enter the second word: in
Ida is great

**12.** Write a Python program to Capitalize the first character of each word in a String 

**<font color ='red'>[You cannot use the built-in upper() function]</font>**

**Sample Input:**

I love python programming

**Sample Output:**

I Love Python Programming

In [5]:
#todo
sentence=input('Enter the sentence:').split(' ')
list1=[]

for i in sentence:
    x= ord(i[0])
    if(x >= 97 and x <= 122):
        y= chr(x-32)
        text= i.replace(i[0],y)
        list1.append(text)
    else:
        list1.append(i)

for k in list1:
    print(k,end=' ')

Enter the sentence:I love python programming
I Love Python Programming 

**13.** Write a Python program that takes a list of numbers separated by comma(“,”) as input of a List
A(Without using loop).Multiply the numbers and print the product.

**<font color='red'>[Hint:You can use split()]**

**Sample Input:**

1,2,3,4,5

**Sample output:**

Product :120

In [8]:
#todo
numbers=input('Enter the numbers: ')
list1=numbers.split(',')

product=1
for i in list1:
    x=int(i)
    product= product*x

print(product)

Enter the numbers: 1,2,3,4,5
120


**14.** Write a Python program that takes two lists from the user. Print both lists simultaneously such that
list1 should display items in original order and list2 in reverse order.

**Sample Input:**

List1 = [10,20,30,40]

List2 = [100,200,300,400]

**Sample Output:**

10 400

20 300

30 200

40 100

In [13]:
#todo
numbers1=input('Enter the numbers: ')
list1=numbers1.split(',')
numbers2=input('Enter the numbers: ')
list2=numbers2.split(',')
list2.reverse()

for i in range (0,len(list1)):
    print(list1[i],list2[i])

Enter the numbers: 10,20,30,40
Enter the numbers: 100,200,300,400
10 400
20 300
30 200
40 100


**15.** Write a Python program to reverse strings in a given list of string values.

**Sample Input:**

['Red','Green','Blue','White','Black']

**Sample Output:**

['deR', 'neerG', 'eulB', 'etihW', 'kcalB']

In [4]:
#todo
list1=[]
num=int(input('Enter the numbers of colors: '))
for i in range(0,num):
    x=input('Enter the colors: ')
    list1.append(x)
    
list2=[]

for i in list1:
    ele = i[::-1]
    list2.append(ele)

print(list2)

Enter the numbers of colors: 5
Enter the colors: red
Enter the colors: white
Enter the colors: black
Enter the colors: blue
Enter the colors: green
['der', 'etihw', 'kcalb', 'eulb', 'neerg']


**16.** Write a Python program that takes a list from the user .Then find the first even and first odd number in the list .

**Sample Input:**

[1, 3, 5, 7, 4, 1, 6, 8]

**Sample Output:**

First Even:4 and First Odd:1

In [9]:
#todo
numbers=input('Enter the numbers: ')
list1=numbers.split(',')

even=0
odd=0
for i in list1:
    x=int(i)
    if(x%2 == 0):
        even=x
    else:
        odd=x
    break

print('First Even: ',even,'and First Odd: ',odd)

Enter the numbers: 4,6,78
First Even:  4 and First Odd:  0


**17.** Write a Python program that takes a list A and a number as input from the user .Create a list by
concatenating the input list which range goes from 1 to n.

**Sample input:**

A = ['p', 'q']

N = 5

**Sample Output:**

['p1', 'q1', 'p2', 'q2', 'p3', 'q3', 'p4', 'q4', 'p5', 'q5']

In [14]:
#todo
numbers=input('Enter the element: ')
list1=numbers.split(',')

num=int(input('Enter the number: '))

list2=[]
for k in range(1,num+1):
    for i in list1:
        x = i + str(k)
        list2.append(x)
        
print(list2)    

Enter the element: p,q
Enter the number: 5
['p1', 'q1', 'p2', 'q2', 'p3', 'q3', 'p4', 'q4', 'p5', 'q5']


# <font color= 'purple'>-----------------------------------------------------------------------------------------------</font>
# <font color= 'purple'>-----------------------------------------------------------------------------------------------</font>
#  

**PROBLEM 1.** Number Line

**Sample input:**

6

**Sample output:**

123456

In [16]:
#todo


**PROBLEM 2.** Star Line
Print as many stars as given in input

**Sample input:**

6

**Sample output:**

    ******

In [17]:
#todo



**PROBLEM 3.** Rectangle

**Sample input:**

4

6

**Sample output:**
    
    ******
    ******
    ******
    ******
#### Hint: 4 and 6 means 4 lines of stars having 6 stars in each line.

In [18]:
#todo


**PROBLEM 4.** Rectangle

**Sample input:**

4

6

**Sample output:**

123456\
123456\
123456\
123456

#### Hint: 4 and 6 means 4 lines of numbers having 1..6 in each line.

In [19]:
#todo


**PROBLEM 5.** \
Triangle - Left Justified\
Draw right angled triangle of given height

**Sample input:**

4

**Sample output:**
       
    *
    **
    ***
    ****

#### Hint: One loop for lines, another loop for printing i number of starts when it is line i.

In [20]:
#todo


**PROBLEM 6.** \
Triangle - Left Justified\
Draw right angled triangle of given height

**Sample input:**

4

**Sample output:**

1\
12\
123\
1234

In [21]:
#todo


**PROBLEM 7.**\
Triangle - Right Justified\
Draw right angled triangle of given height

**Sample input:**

4

**Sample output:**

       *
      **
     ***
    ****


#### Hint: Count and print appropriate number of spaces in front of stars. Notice that there is one less space and one more star in each line.

In [22]:
#todo


**PROBLEM 8.**\
Triangle - Right Justified\
Draw right angled triangle of given height

**Sample input:**

4

**Sample output:**

       1
      12
     123
    1234

In [23]:
#todo


**PROBLEM 9.**\
Triangle - Isosceles\
Draw triangle of given height

**Sample input 1:**

3

**Sample output 1:**

      *
     ***
    *****

**Sample input 2:**

4

**Sample output 2:**

       *
      ***
     *****
    *******

In [24]:
#todo


**PROBLEM 10.**\
Triangle - Isosceles\
Draw triangle of given height

**Sample input 1:**

3

**Sample output 1:**

      1
     123
    12345

**Sample input 2:**

4

**Sample output 2:**

       1
      123
     12345
    1234567

In [25]:
#todo


**PROBLEM 11.** \
Triangle - Right Justified\
Draw right angled triangle of given height

**Sample input:**

4

**Sample output:**

       4
      34
     234
    1234

In [26]:
#todo


**PROBLEM 12.**  
Rhombus\
Just draw the image of the above triangle once. And then, the opposite, once.

**Sample input:**

3

**Sample output:**

      *
     ***
    *****
     ***
      *

In [27]:
#todo


**PROBLEM 13.**

Rhombus\
Just draw the image of the above triangle once. And then, the opposite, once.

**Sample input:**

3

**Sample output:**

      1
     123
    12345
     123
      1

In [28]:
#todo



**PROBLEM 14.** \
Hollow Rectangle\
Display a rectangle of given length and width.

**Sample input:**

4

5

**Sample output:**

    *****
    *   *
    *   *
    *****

#### Hint 1: Print the character space (‘ ‘) in the middle.

#### Hint 2: You can re-use your solution to PROBLEM 2 and use if condition to selectively print first and last star of 
#### each line and all stars of first and last line.

In [29]:
#todo


**PROBLEM 15.**\
Hollow Rectangle\
Display a rectangle of given length and width.

**Sample input:**

4

5

**Sample output:**

    12345
    1   5
    1   5
    12345

In [30]:
#todo


**PROBLEM 16.** \
Hollow Triangle - Left Justified\
Draw right angled triangle of given height

**Sample input:**

5

**Sample output:**

    *
    **
    * *
    *  *
    *****

In [31]:
#todo


**PROBLEM 17.**\
Hollow Triangle - Left Justified\
Draw right angled triangle of given height

**Sample input:**

5

**Sample output:**

    1
    12
    1 3
    1  4
    12345

In [32]:
#todo


**PROBLEM 18.** 

Hollow Triangle\
Draw right angled triangle of given height

**Sample input:**

5

**Sample output:**

        *
       **
      * *
     *  *
    *****

In [33]:
#todo


**PROBLEM 19.**\
Hollow Triangle - Right Justified\
Draw right angled triangle of given height

**Sample input:**

5

**Sample output:**

        5
       45
      3 5
     2  5
    12345

In [34]:
#todo


**PROBLEM 20.** \
Hollow Triangle\
Draw triangle of given height

**Sample input 1:**

3

**Sample output 1:**

      *
     * *
    *****

**Sample input 2:**

4

**Sample output 2:**

       *
      * *
     *   *
    *******

In [35]:
#todo


**PROBLEM 21.**\
Hollow Triangle – Isosceles\
Draw triangle of given height

**Sample input 1:**

3

**Sample output 1:**

      1
     1 3
    12345

**Sample input 2:**

4

**Sample output 2:**

       1
      1 3
     1   5
    1234567

In [36]:
#todo


**PROBLEM 22.** 

Hollow Rhombus\
Just draw the image of the above triangle once. And then, the opposite, once.

**Sample input:**

3

**Sample output:**

      *
     * *
    *   *
     * *
      *

In [37]:
#todo


**PROBLEM 23.** \
Hollow Rhombus\
Just draw the image of the above triangle once. And then, the opposite, once.

**Sample input:**

3

**Sample output:**

      1
     1 3
    1   5
     1 3
      1

In [38]:
#todo


**PROBLEM 24.**\
Palindrome

**Sample input:**

5

**Sample output:**
    
123454321

In [ ]:
#todo



**PROBLEM 25.**\
Palindromic Triangle

**Sample input:**

5

**Sample output:**

        1
       121
      12321
     1234321
    123454321

In [39]:
#todo
